axlearn/common/config.py

```diff
@@ -371,7 +371,14 @@ def to_flat_dict(self, *, omit_default_values: Collection[Any]) -> Dict[str, Any
         result = {}
 
         def enter(key: str, val: Any, default_result: Optional[List]) -> Optional[List]:
-            if key and isinstance(val, ConfigBase):
+            if dataclasses.is_dataclass(val) and not isinstance(val, type):
+                omitted_result = []
+                for cur_key, cur_val in default_result:
+                    if cur_val in omit_default_values:
+                        continue
```

Comment:
We should omit only if cur_val is the default value of the dataclass field, e.g.,

```
@dataclass 
class foo:
    x: Optional[int] = None
    y: Optional[int] = -1
```

We should omit if `x is None` but not if `y is None`. This is to make it consistent with the attribute values below.

Reply:
made a change and added a test like this